packages/vite/src/node/server/index.ts

```diff
@@ -366,7 +366,21 @@ export async function createServer(
     },
     ssrFixStacktrace(e) {
       if (e.stack) {
-        e.stack = ssrRewriteStacktrace(e.stack, moduleGraph)
+        const stacktrace = ssrRewriteStacktrace(e.stack, moduleGraph)
+        const { configurable, writable } = Object.getOwnPropertyDescriptor(
+          e,
+          'stack'
+        )!
+        if (configurable) {
+          Object.defineProperty(e, 'stack', {
+            value: stacktrace,
+            enumerable: true,
+            configurable: true,
+            writable: true
+          })
+        } else if (writable) {
+          e.stack = stacktrace
+        }
```

Comment:
Should we extract this into a helper? Maybe
```suggestion
        const stacktrace = ssrRewriteStacktrace(e.stack, moduleGraph)
        rebindErrorStacktrace(e,stacktrace)
```

Reply:
Seems reasonable, yeah. I couldn't think of a name so I'll just use `rebindErrorStacktrace` as suggested.